ui/src/main/webapp/src/app/validators/file-exists.validator.ts

```diff
@@ -1,12 +1,12 @@
-import {AbstractControl} from "@angular/forms";
+import {AbstractControl, AsyncValidatorFn} from "@angular/forms";
 import {FileService} from "../services/file.service";
 
 export class FileExistsValidator {
 
-    static create(fileService: FileService) {
-        return function (c: AbstractControl): {[key: string]: any} {
+    static create(fileService: FileService): AsyncValidatorFn {
+        return function (control: AbstractControl): {[key: string]: any} {
             return new Promise(resolve => {
-                fileService.pathExists(c.value).subscribe(result => {
+                fileService.pathExists(control.value).subscribe(result => {
```

Comment:
Right... the function definition has c so I guess IDEA uses that but I like descriptive code